tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -43,22 +43,28 @@ public void ActivityStarted<T>(string sourceName, T activity)
             ulong? spanId = null;
             string? rawTraceId = null;
             string? rawSpanId = null;
+            bool isRemoteContext = false;
 
             if (activity is IW3CActivity w3cActivity)
             {
                 // If the user has specified a parent context, get the parent Datadog SpanContext
                 if (w3cActivity.ParentSpanId is not null
-                    && w3cActivity.ParentId is string parentId
-                    && ActivityMappingById.TryGetValue(parentId, out ActivityMapping mapping))
+                 && w3cActivity.ParentId is { } parentId)
                 {
-                    parent = mapping.Scope.Span.Context;
+                    if (ActivityMappingById.TryGetValue(parentId, out ActivityMapping mapping))
+                    {
+                        parent = mapping.Scope.Span.Context;
+                    }
+                    else
+                    {
+                        // create a new parent span context for the ActivityContext
```

Comment:
I wonder if adding the example from the PR description could help future reads of this